modules/fbs/includes/fbs.reservation.inc

```diff
@@ -12,95 +12,116 @@
  *   User to list reservations for.
  * @param string|null $requested_type
  *   Type of reservations to list. Null for all.
+ * @param bool $reset
+ *   If TRUE cache is reset (default: FALSE)
  *
  * @return array
  *   Reservation info.
  */
-function fbs_reservation_list($account, $requested_type = NULL) {
-  $res = array();
-  try {
-    $res = fbs_service()->Reservations->getReservations(fbs_service()->agencyId, fbs_patron_id($account));
-  }
-  catch (Exception $e) {
-    watchdog_exception('fbs', $e);
-  }
-
-  $result = array(
-    DING_RESERVATION_READY => array(),
-    DING_RESERVATION_NOT_READY => array(),
-    DING_RESERVATION_INTERLIBRARY_LOANS => array(),
-  );
+function fbs_reservation_list($account, $requested_type = NULL, $reset = FALSE) {
+  $results = &drupal_static(__FUNCTION__, array());
 
-  $ids = array();
-  foreach ($res as $reservation) {
-    $ids[$reservation->recordId] = $reservation->recordId;
-  }
-  $ids = ding_provider_build_entity_id($ids);
-
-  foreach ($res as $reservation) {
-    $type = $reservation->state == 'readyForPickup' ? DING_RESERVATION_READY : DING_RESERVATION_NOT_READY;
-    $type = $reservation->state == 'interLibraryReservation' ? DING_RESERVATION_INTERLIBRARY_LOANS : $type;
-
-    $ding_reservation = array(
-      'ding_entity_id' => $ids[$reservation->recordId],
-      'id' => $reservation->reservationId,
-      'pickup_branch_id' => $reservation->pickupBranch,
-      'created' => $reservation->dateOfReservation,
-      'expiry' => $reservation->expiryDate,
-      // Required by ding_reservation.
-      'reservation_type' => $type,
-    );
-
-    if ($type === DING_RESERVATION_READY) {
-      $ding_reservation['pickup_date'] = $reservation->pickupDeadline;
-      $ding_reservation['pickup_order_id'] = $reservation->pickupNumber;
-      $ding_reservation['order_arrived'] = TRUE;
-    }
-    elseif ($type === DING_RESERVATION_NOT_READY || $type === DING_RESERVATION_INTERLIBRARY_LOANS) {
-      if (!empty($reservation->numberInQueue)) {
-        $ding_reservation['queue_number'] = $reservation->numberInQueue;
-      }
+  if (empty($results) || $reset) {
+    // Check if ding_session_cache is available.
+    if (module_exists('ding_session_cache') && !$reset) {
+      $results = ding_session_cache_get('fbs', 'reservation', array());
     }
 
-    // Add in periodical information if it is one.
-    if ($reservation->periodical) {
-      $periodical = $reservation->periodical;
-      $vol = $periodical->volume;
-      if (!empty($periodical->volumeNumber)) {
-        $vol .= '.' . $periodical->volumeNumber;
+    // If both static cache and session cache failed, try reloading information
+    // from FBS.
+    if (empty($results) || $reset) {
```

Comment:
As above.

Reply:
Changed default value to NULL and using is_null to check it.